src/02-common-principles.md

```diff
@@ -105,36 +102,62 @@ misunderstanding we clarify them here.
+    **file extension**; for example, it is `eeg` in `sub-05_task-matchingpennies_eeg.vhdr`.
 
 1.  **File extension** - a portion of the file name after the left-most
     period (`.`) preceded by any other alphanumeric. For example, `.gitignore` does
     not have a file extension, but the file extension of `test.nii.gz` is `.nii.gz`.
     Note that the left-most period is included in the file extension.
 
-1.  **DEPRECATED** - A "deprecated" entity or metadata field SHOULD NOT be used in the
+1.  **DEPRECATED** - A "deprecated" [entity](#entities) or metadata field SHOULD NOT be used in the
     generation of new datasets.
     It remains in the standard in order to preserve the interpretability of existing datasets.
     Validating software SHOULD warn when deprecated practices are detected and provide a
     suggestion for updating the dataset to preserve the curator's intent.
 
+## Entities
+
+An "entity" is an attribute that can be associated with a file, contributing
+to the identification of that file as a component of its file name in the
+form of a hyphen-separated key-value pair.
+
+Each entity has the following attributes:
+
+1.  *Name*: A comprehensive name describing the context of information
+    to be provided via the entity.
+1.  *Key*: A short string, typically a compression of the entity name,
+    which uniquely identifies the entity when part of a file name.
+1.  *Value type*: The requisite form of the value that gets specified
+    alongside the key whenever the entity appears in a file name.
+    For each entity, the value is of one of two possible types:
+    1.  *Index*: A non-negative integer, potentially zero-padded for
+        consistent width.
+    1.  *Label*: An alphanumeric string.
+        Note that labels MUST not collide when casing is ignored
+        (see [Case collision intolerance](#case-collision-intolerance)).
+
+The entity *format* is a string that prescribes how the entity appears within
+any given file name. For a hypothetical entity with key "`key`", the format
+can be either "`key-<index>`" or "`key-<label>`", depending on the value
+type of that entity.
+
+An entity *instance* is the specific manifestation of an entity within the
+name of a specific file, based on the format of the entity but with a value
+that provides identifying information to the particular file in whose name
+it appears.
+
+Depending on context, any one of the entity name, key, format, or a specific
+entity instance, may in some instances be referred to as simply an "entity".
+
+"Subject", "session", "sample", "task", and "run" from the list of definitions
+above are all examples of entities. The comprehensive list of supported
+entities is defined in [Appendix IX](99-appendices/09-entities.md);
+further, whether each is OPTIONAL, REQUIRED, or MUST NOT be provided for
+various data files, as well as their relative ordering in a file name, are
+defined in the Entity Table in [Appendix IV](99-appendices/04-entity-table.md).
```

Comment:
line breaks and added a word "compulsory"

```suggestion
various data files, as well as their compulsory relative ordering in a file name,
are defined in the Entity Table in
[Appendix IV](99-appendices/04-entity-table.md).
```

Reply:
Closed by 3972dd7.